galleries/examples/color/colorbar_histogram.py

```diff
@@ -0,0 +1,64 @@
+"""
+=========================
+Histogram as Colorbar
+=========================
+
+This example demonstrates how to create a colorbar for an image and
+add a histogram of the data values alongside it. This is useful for
+visualizing the distribution of values mapped to colors.
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.colors as mcolors
+from matplotlib.ticker import MaxNLocator
+
+# === Surface Data ===
+delta = 0.025
+x = y = np.arange(-2.0, 2.0, delta)
+X, Y = np.meshgrid(x, y)
+Z1 = np.exp(-((X+1)*1.3)**2 - ((Y+1)*1.3)**2)
+Z2 = 2.5*np.exp(-(X - 1)**2 - (Y - 1)**2)
+Z = (Z1**0.25 - Z2**0.5)
+
+# === Histogram from actual Z data ===
+counts, bins = np.histogram(Z, bins=30)
+
+# === Colormap & Normalization ===
+cmap = plt.get_cmap('RdYlBu_r')
+norm = mcolors.BoundaryNorm(bins, cmap.N)
+
+# === Main Plot ===
+fig, ax = plt.subplots()
+im = ax.imshow(Z, cmap=cmap,
+               origin='lower', extent=[-3, 3, -3, 3],
+               norm=norm)
+
+# Adjust image position to allow space
+plt.subplots_adjust(right=0.78, top=0.92, bottom=0.08)
+
+# === Inset Histogram – Positioning adjusted ===
+cax = ax.inset_axes([1.18, 0.02, 0.25, 0.95])  # left, bottom, width, height
```

Comment:
Our of curiosity: Wouldn't it be simpler to use `plt.subplots(1, 2, width_ratios=...)` instead of manually adjusting the axes?